### PR TITLE
automate todo-list build

### PR DESCRIPTION
**Pull request by shihchang**

### CODE REVIEW

File: todo-list/deploy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ENV DOMAIN_HOME=${DOMAIN_PARENT}/${DOMAIN_NAME} \
+    PROPERTIES_FILE_DIR=$ORACLE_HOME/properties \
+    WDT_HOME="/u01" \
+    SCRIPT_HOME="${ORACLE_HOME}" \
+    PATH=$PATH:${ORACLE_HOME}/oracle_common/common/bin:${ORACLE_HOME}/wlserver/common/bin:${DOMAIN_HOME}:${DOMAIN_HOME}/bin:${ORACLE_HOME}
+
+COPY weblogic-deploy.zip ${WDT_HOME}
+COPY container-scripts/* ${SCRIPT_HOME}/
+
+# Create the properties file directory and the domain home parent with the correct permissions / owner.
```
Comment (wmhopkins) [inline todo-list/deploy/Dockerfile:52]: Previously, we simply copied the model into the domain, now we're building an image in the domain. Why the change?

File: todo-list/deploy/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+VOLUME $DOMAIN_HOME
+VOLUME $WDT_HOME
+
+# Expose admin server, managed server port and domain debug port
+EXPOSE $ADMIN_PORT $MANAGED_SERVER_PORT $DEBUG_PORT
+
+WORKDIR $DOMAIN_HOME
+
+# Define default command to start Admin Server in a container.
+CMD ["/u01/oracle/startAdminServer.sh"]
```
Comment (wmhopkins) [inline todo-list/deploy/Dockerfile:115]: Leaving this as it was previously would provide for a 10 minute sleep, to allow someone to log in and look around if things didn't work right. The WebLogic operator will ensure that the command is overridden to start WebLogic if the domain creation succeeds.

File: todo-list/deploy/build.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
```
Comment (wmhopkins) [inline todo-list/deploy/build.sh:1]: Most of what's in here seems unnecessary -- won't "docker build", with the given Dockerfile, achieve the same result?

File: todo-list/deploy/container-scripts/startAdminServer.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
```
Comment (wmhopkins) [inline todo-list/deploy/container-scripts/startAdminServer.sh:1]: Won't WDT and the WebLogic operator take care of all of this stuff automatically? Why do we need to supply these scripts?

File: todo-list/deploy/exporter-config.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  - persistentStoreRuntimes:
+      prefix: wls_persistentstore_
+      key: name
+  - JDBCServiceRuntime:
+      JDBCDataSourceRuntimeMBeans:
+        prefix: wls_datasource_
+        key: name
+  - JTARuntime:
+      prefix: wls_jta_
+      key: name
```
Comment (wmhopkins) [inline todo-list/deploy/exporter-config.yaml:53]: needs newline.

**Review by wmhopkins — COMMENTED** [review]
Hi Chen -- there seem to be a lot of changes here that to change the structure of the app and how it's built. I understand changing the name of the setup folder, for example, and moving things around to align with other apps (I assume), but some of the changes, like building the domain in the image, rather than just placing a WDT model file, and providing a lot of customized weblogic start scripts, seem gratuitous.

File: todo-list/deploy/Dockerfile (hunk 1 above)
Comment (shihchang) [reply todo-list/deploy/Dockerfile]: changed to use original setup/Dockerfile

File: todo-list/deploy/build.sh (hunk 3 above)
Comment (shihchang) [reply todo-list/deploy/build.sh]: changed to use original setup/Dockerfile

File: todo-list/deploy/container-scripts/startAdminServer.sh (hunk 4 above)
Comment (shihchang) [reply todo-list/deploy/container-scripts/startAdminServer.sh]: removed this file

File: todo-list/deploy/Dockerfile (hunk 2 above)
Comment (shihchang) [reply todo-list/deploy/Dockerfile]: changed to use original setup/Dockerfile

File: todo-list/deploy/exporter-config.yaml (hunk 5 above)
Comment (shihchang) [reply todo-list/deploy/exporter-config.yaml]: fixed

**Review by wmhopkins — APPROVED** [review]
Looks good to me. Only thing I might change is the name of build.sh, to make it more consistent with the names of the other scripts, but that's a trivial issue. Approved.
